packages/aws-cdk-lib/aws-lambda/lib/function.ts

```diff
@@ -1149,6 +1160,33 @@ Environment variables can be marked for removal when used in Lambda@Edge by sett
     this.addEnvironment('AWS_LAMBDA_EXEC_WRAPPER', props.adotInstrumentation.execWrapper);
   }
 
+  /**
+   * Add a Parameters and Secrets Extension Lambda layer.
+   */
+  private configureParamsAndSecretsExtension(props: FunctionProps): void {
+    if (props.paramsAndSecrets === undefined) {
+      return;
+    }
+
+    // grant permissions to lambda execution role to allow access to provided secrets
+    props.paramsAndSecrets.secrets?.forEach(secret => {
+      secret.grantRead(this);
+      if (secret.encryptionKey) {
+        secret.encryptionKey.grantDecrypt(this);
```

Comment:
Feels like `secret.grantRead()` should grant this automatically. Does it not?

Reply:
Great catch. Just checked the code and it does grant decrypt as part of grantRead when an encryptionKey is specified.